docs/src/image.md

```diff
@@ -0,0 +1,20 @@
+# Image type syntax
+
+There are a huge number of combinations of image types in SPIR-V. They are represented by a const
+generic type called `spirv_std::image::Image`, however, specifying the generic parameters of this
+type is incredibly tedious, so a wrapper macro, `spirv_std::Image!` can be used to write the type
+instead.
+
+The specific syntax and meaning of the arguments to the `Image!` macro can be found in
+[rustdoc](https://embarkstudios.github.io/rust-gpu/api/spirv_std/macro.Image.html).
+
+Some type aliases for common image formats can be found in the `spirv_std::image` module. For
```

Comment:
Could this also link to the module docs? i.e. something like
```suggestion
Some type aliases for common image formats can be found in the [`spirv_std::image`](https://embarkstudios.github.io/rust-gpu/api/spirv_std/index.html) module. For
```

However, something strange is afoot. [That page](https://embarkstudios.github.io/rust-gpu/api/spirv_std/index.html) lists the version as `0.4.0-alpha.9`, the latest version. However, it does not list the `image` module as a seperate module, and lists all the image types as types at the root.

Also, the [src] of `Image2d` is [shown as this](https://embarkstudios.github.io/rust-gpu/api/src/spirv_std/textures.rs.html#19-21), which is afaik not correct.

Reply:
This is because the docs are not generated with the `const_generics` feature, and so the new `Image2d` doesn't show up (the new one is in the `image` module)

Reply:
Oh, I see sorry. That also solves my other confusion.